layers.go

```diff
@@ -160,6 +160,9 @@ type Layer struct {
 	// convenience of the caller.  They can be large, and are only in
 	// memory when being read from or written to disk.
 	BigDataNames []string `json:"big-data-names,omitempty"`
+
+	// Incremental is true if this layer was created by a partial pull.
+	Incremental bool `json:"incremental,omitempty"`
```

Comment:
“was created by a partial pull” does not help consumers of the value much to understand what needs doing. Does this mean “if `UncompressedDigest` is not actually an uncompressed digest”?

Basically this would force every user of `store.LayersByUncompressedDigest` to now start checking `Incremental` on the result to see whether calling `.Diff` is safe.

That _might_ be good enough, but it also feels unclean.

Alternatively, we could track _exactly_ what we know, set `.UncompressedDigest` to `""`, and to add a new `ChunkedTOCDigest` field (and a new `LayersByChunkedTOCDigest` function). That seems more correct, OTOH also probably triggering more invasive changes in callers.


---

Either way, c/image’s `TryReusingBlob*` would need to be taught about TOC digests, and trigger a search by the TOC digest value. … And at which layer would we turn the blob annotation in to a TOC digest?

Do we need, instead of `LayersByChunkedTOCDigest`, some kind of `ChunkedLayersByManifestAnnotations`?